### PR TITLE
Travis Ci(unix and osx build) and AppVeyor(windows) badges in Readme.md

### PR DESCRIPTION
**Pull request by boddicheg**
Add badges like this:

[Build Status](https://travis-ci.org/boddicheg/glfw) [Build Status](https://ci.appveyor.com/project/boddicheg/glfw)

to Readme.md with current build status on osx, linux and windows.
If this pull request will be accepted, after every commit in master branch this badges will be updated according current build status.

[Example](https://github.com/boddicheg/glfw/tree/multi-platform)

P.S: both badges are clickable
Thanks!

### CODE REVIEW

**Comment by shurcooL** [comment]
CI is pretty useful, nice job on making it work.

It's fairly basic in that it only tests that the code builds successfully (there are no automatic tests). But it's still useful to prove that the CMake setup and such is resulting in something than can build successfully.

File: install-deps.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+wget http://www.cmake.org/files/v2.8/cmake-2.8.12.tar.gz
+tar xzf cmake-2.8.12.tar.gz
+cd cmake-2.8.12
+cmake .
+make
+sudo make install
```
Comment (linkmauve) [inline install-deps.sh]: You probably don’t need to be root to install this package, just set its prefix to something like `~/.local`.

In Citra we download the binary instead, to make the build shorter, maybe you could take inspiration from this script: https://github.com/citra-emu/citra/blob/master/.travis-deps.sh

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # GLFW
-
+[![Build Status](https://travis-ci.org/boddicheg/glfw.svg?branch=multi-platform)](https://travis-ci.org/boddicheg/glfw)
+[![Build Status](https://ci.appveyor.com/api/projects/status/github/boddicheg/glfw?branch=multi-platform&svg=true)](https://ci.appveyor.com/project/boddicheg/glfw)
```
Comment (linkmauve) [inline README.md]: You should leave two blank lines here, to make this file more readable.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+language: cpp
+
+os:
+- linux
+- osx
+
+sudo: true
+
+before_script:
+- sh install-deps.sh
```
Comment (linkmauve) [inline .travis.yml]: If you put the executable bit on this file, and a shebang, you don’t have to hardcode the interpreter to use.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+script:
+- cmake .
```
Comment (linkmauve) [inline .travis.yml]: Why don’t you use the same method as appveyor, that is to build from a `build` directory?

**Comment by linkmauve** [comment]
It would be even more useful to also run the tests, but I don’t think either of the providers you chose support that.

Aside from my few remarks, LGTM.

**Comment by elmindreda** [comment]
The union of the changeset looks good but please squash it.

**Comment by elmindreda** [comment]
Travis and Appveyor CI has been added.

**Comment by shurcooL** [comment]
Nice!

I want to clarify what the travis CI test checks for. Is it only checking that the source code builds successfully without any errors? Does it also run any tests? Does it check against any runtime problems (like `glfwCreateWindow` failing to create a window)?

**Comment by boddicheg** [comment]
@shurcooL 

> Is it only checking that the source code builds successfully without any errors?

Yep

>Does it also run any tests?

Nope, but if you add some script to ```script``` path to  .travis.yml file like this:

```
cmake tests/ 
```

so then it will.

> Does it check against any runtime problems (like glfwCreateWindow failing to create a window)?

I think it does but i am not sure.

**Comment by elmindreda** [comment]
@shurcooL This tests compilation and linking only. No GLFW code is run, but this is still very helpful.

The current set of tests aren't suited for offline testing, but with this merged it's much more attractive to write ones that are. I don't believe most of GLFW can be usefully tested offline, but the bits that can should be.

**Pull request closed without merging**